test/integration/idle_timeout_integration_test.cc

```diff
@@ -284,19 +304,34 @@ TEST_P(IdleTimeoutIntegrationTest, RequestTimeoutIsNotDisarmedByEncode100Continu
   enable_request_timeout_ = true;
 
   auto response = setupPerStreamIdleTimeoutTest("POST");
-  upstream_request_->encode100ContinueHeaders(Http::TestHeaderMapImpl{{":status", "100"}});
+  upstream_request_->encode100ContinueHeaders(
+      Http::TestHeaderMapImpl{{":status", std::to_string(enumToInt(Http::Code::Continue))}});
 
   waitForTimeout(*response, "downstream_rq_timeout");
 
   EXPECT_FALSE(upstream_request_->complete());
   EXPECT_EQ(0U, upstream_request_->bodyLength());
   EXPECT_TRUE(response->complete());
-  EXPECT_STREQ("408", response->headers().Status()->value().c_str());
+  EXPECT_STREQ(std::to_string(enumToInt(Http::Code::RequestTimeout)).c_str(),
+               response->headers().Status()->value().c_str());
   EXPECT_EQ("request timeout", response->body());
 }
 
-// TODO(auni53) create a test filter that hangs and does not send data upstream, which would
-// trigger a configured request_timer
+TEST_P(IdleTimeoutIntegrationTest, RequestTimeoutIsNotDisarmedByASlowFilter) {
```

Comment:
I think what you're testing here is not "RequestTimeoutIsNotDisarmed" (which is a codepath in which the request timeout is NOT triggered but remains active), but rather something like "RequestTimeoutTriggersOnSleepingFilter".

nit: SleepingFilter over ASlowFilter, to be in line w variable naming

Reply:
I agree with the first suggestion. I chose `SlowFilter` to explain the behavior of the test and not the underlying implementation that causes it (which in this case happens to be the `SleepingFilter`).